### ISSUE TITLE
Add translation Example to Ignite

### ISSUE THREAD

**Issue by Ishan-Kumar2** [pytorch/ignite#2172]
## :rocket: Feature
Ignite already has a example for Text Classification using Huggingface Transformers, but having a example with sequence to sequence model used for Translation like [this](https://github.com/huggingface/transformers/blob/master/examples/pytorch/translation/run_translation.py) would be useful. This is because unlike the classification example here the output of the model is a sequence so there are changes in all the functions. The [BART model](https://arxiv.org/abs/1910.13461) can be added.

This example would be useful for anyone implementing Translation, Question Answering and Summarisation and other seq2seq tasks. Through this, we can also show an example of how to use BLEU and Rouge metrics present in ignite.
Could do this in the form of a notebook, or script with more features and ability to extend.

**Comment by vfdev-5** [comment]
This is great suggestion, @Ishan-Kumar2 , thanks!
@Priyansi can you please explain how to proceed with such a how-to guide or tutorial.

**Comment by Priyansi** [comment]
Hey, @Ishan-Kumar2 thank you for the suggestion! We are currently working on making tutorials and how-to guides for Ignite in this [repo](https://github.com/pytorch-ignite/examples). All the content is in the form of notebooks.
I am going to work on converting the Text Classification example with transformers that you mentioned to a notebook with explanation. And I agree, having a tutorial on translation would be a great NLP tutorial with Ignite.

Would you be willing to contribute and make this tutorial possible? You can simply clone the repo, make a folder named `tutorials1 and add your notebook there.

**Comment by Ishan-Kumar2** [comment]
@Priyansi sounds good, although are you sure we should go for a notebook? I think having python scripts may be easier to make changes to. Let me know your thoughts :)

**Comment by Priyansi** [comment]
Yes, we decided to go with notebooks because readers would find it more intuitive to see the output as they read along plus it cuts down on all unnecessary code and maintains a flow. It's also so we can convert these notebooks to markdown for easy hosting on our site.

We are trying to organize our examples into tutorials and how-to guides. Since this example's objective, as you stated above, would be to walk through **various** Ignite concepts concerning NLP, this example would be better as a tutorial.

cc @vfdev-5
